packages/components/src/presentation-components/properties/inputs/AsyncSelect.tsx

```diff
@@ -144,8 +172,12 @@ export function AsyncSelect<OptionType, Group extends GroupBase<OptionType>, Add
           SelectContainer,
           NoOptionsMessage,
         }}
+        selectRef={selectRef}
+        onMenuOpen={onMenuOpen}
+        onMenuClose={onMenuClose}
+        menuPortalTarget={divRef.current?.ownerDocument.body.querySelector(".presentation-instance-filter-dialog")}
```

Comment:
I think we need to allow specifying different portal target instead of hard coding selector, because it can be used in property-grid too. Also in other spaces where editors from `@itwin/components-react` are used. Maybe we could add React context for specifying portal target.

Reply:
Added a context for providing portal targets.